CDP4ServicesDal/CdpServicesDal.cs

```diff
@@ -89,13 +89,29 @@ public class CdpServicesDal : Dal
         /// </summary>
         private HttpClient httpClient;
 
+        /// <summary>
+        /// Asserts that the MessagePack deserialization should be used or not
+        /// </summary>
+        private readonly bool isMessagePackSupported;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="CdpServicesDal"/> class.
         /// </summary>
-        public CdpServicesDal()
+        public CdpServicesDal(): this(true)
         {
-            this.Cdp4JsonSerializer = new Cdp4JsonSerializer(this.MetaDataProvider, this.DalVersion);
+            // Default constructor
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CdpServicesDal"/> class.
+        /// </summary>
+        /// <param name="isMessagePackSupported">Asserts that the MessagePack deserialization should be used or not. Supported by default</param>
+        public CdpServicesDal(bool isMessagePackSupported)

```

Comment:
I don;t understand why this option is necessary. The CDPServicesDAal shall always support messagepack, i see no reason to disable this. Can you explain?

Reply:
This was introduced by Antoine. Don't know the exact details. @antoineatstariongroup could you explain?

Reply:
As far as I remember, is to allow the flexibility to use add the header for messagepack realization or not. By default, it's supported but we could force the server to not use messagepack as serializer. Just there to put some flexibility. We had a discussion about that while I was working on it

Reply:
I prefer to revert this. By using HTTP Accept header we ask the server to respond in a certain way. We don't want to force the server, we just ask  "if you know of MessagePack, please respond with MEssagePack"

Reply:
i still see this in the code, please revert. The Dal figures it out and its not something I want to come from the caller.